src/webgpu/shader/validation/expression/call/builtin/const_override_validation.ts

```diff
@@ -100,10 +112,15 @@ export function validateConstOrOverrideBuiltinEval(
 
   switch (stage) {
     case 'constant': {
+      let val_str = value.toString();
+      if (!val_str.includes('.') && !val_str.includes('e')) {
```

Comment:
Sadly, JS doesn't have sprintf-like functionality built into the language. I  checked.
We don't want to take on a dependency to sprintf-js.

So this is ok.  It might be fragile if the exponent uses 'E'.  But let's wait until we see that.